### PR TITLE
xzgrep security patch xzgrep-ZDI-CAN-16587

### PR DESCRIPTION
**Pull request by cbouss**
# Changes

Changes:
- Update metadata (https, license)
- Bump build number
- Add security patch xzgrep-ZDI-CAN-16587
  See: https://tukaani.org/xz/ https://www.mail-archive.com/xz-devel@tukaani.org/msg00551.html

# Review Information

## Source

https://git.tukaani.org/

## License

https://git.tukaani.org/?p=xz.git;a=blob_plain;f=COPYING;hb=HEAD

## Upstream Changes

Security patch as explained here: https://tukaani.org/xz/ 

## Issues

NA

## Pins and Requireds

NA

## Testing

Existing tests passed.

# Closing Comments

xz 5.2.5 patched with xzgrep-ZDI-CAN-16587

### CODE REVIEW

**Comment by cbouss** [comment]
prefect failed to download source

**Comment by cbouss** [comment]
concourse: https://concourse.build.corp.continuum.io/teams/main/pipelines/cb-xzgrep

File: recipe/meta.yaml [hunk 1]
```diff
@@ -5,14 +5,18 @@ package:
   version: {{ version }}
 
 source:
-  url: http://downloads.sourceforge.net/project/lzmautils/xz-{{ version }}.tar.bz2
+  url: https://downloads.sourceforge.net/project/lzmautils/xz-{{ version }}.tar.bz2
   sha256: 5117f930900b341493827d63aa910ff5e011e0b994197c3b71c08a20228a42df
   patches:
     - 0001-Add-vs2008-support-via-cmake.patch
     - 0002-Add-gettimeofday-for-VS2008.patch
```
Comment (katietz) [inline recipe/meta.yaml]: Please do not add empty lines here

File: recipe/meta.yaml [hunk 2]
```diff
@@ -43,21 +48,19 @@ test:
     - if not exist %PREFIX%\\Library\\lib\\liblzma.lib exit 1  # [win]
     - if not exist %PREFIX%\\Library\\lib\\liblzma_static.lib exit 1  # [win]
     - if not exist %PREFIX%\\Library\\include\\lzma.h exit 1  # [win]
-    - conda inspect linkages -p $PREFIX $PKG_NAME  # [not win]
```
Comment (katietz) [inline recipe/meta.yaml]: please don't remove those tests.  instead add as requirement 'conda-build' package.  that will fix this test

**Review by katietz — CHANGES_REQUESTED** [review]
otherwise looks great

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml]: Anthony was saying this was replaced by calling --error-overlinking --error-overdepending. Is not the case?

Comment (cbouss) [reply recipe/meta.yaml]: not necessarily true since this is not enabled by default

**Review by ScottBailey — APPROVED** [review]

**Review by varlackc — COMMENTED** [review]
We might need to add the license family to the meta.yaml file
https://github.com/AnacondaRecipes/xz-feedstock/blob/cb_xzgrep_5.2.5_b1/recipe/meta.yaml#L58&&L60

also there seems to be an issue building the package in `prefect` for `osx-64`
https://cloud.prefect.io/anaconda/flow-run/3c87658a-94ee-41a3-8cb5-67f27a9f32c1

other than that the package looks good, we just have to solve those issues and then we can approve the package.

**Comment by cbouss** [comment]
Updated license family. linux-ppc64le is still failing, but again due to sourceforge failed download.

**Review by katietz — APPROVED** [review]
thanks for adding this conda-build requirement for the inspect.

**Pull request merged**